cpp/common/src/codingstandards/cpp/deviations/CodeIdentifierDeviation.qll

```diff
@@ -336,26 +341,60 @@ class CodeIdentifierDeviation extends TCodeIndentifierDeviation {
     )
   }
 
+  predicate hasLocationInfo(
+    string filepath, int suppressedLine, int suppressedColumn, int endline, int endcolumn
+  ) {
+    exists(Comment commentMarker |
+      this = TSingleLineDeviation(_, commentMarker, filepath, suppressedLine) and
+      suppressedColumn = 1 and
+      endline = suppressedLine
+    |
+      if commentMarker instanceof DeviationEndOfLineMarker
+      then endcolumn = commentMarker.(DeviationEndOfLineMarker).getLocation().getEndColumn()
+      else
+        // Find the last column for a location on the next line
+        endcolumn =
+          max(Location l |
+            l.hasLocationInfo(filepath, _, _, _, _) and
+            l.getEndLine() = suppressedLine
+          |
+            l.getEndColumn()
+          )
+    )
+    or
+    this = TMultiLineDeviation(_, _, _, filepath, suppressedLine, endline) and
+    suppressedColumn = 1 and
+    endcolumn = 1
```

Comment:
Some potential inconsistencies. Firstly, the suppressed area includes the text of the first comment, but ends before the text of the second comment.

Additionally, it looks like:

```
  // codeql::misra_deviation_begin(foo)
  non_compliant_code // codeql::misra_deviation_end(foo)
```

neither deviates the `non_compliant_code` nor triggers an invalid deviation warning. The deviated location matched here would not include `non_compliant_code` either.

Do you think either of these are an issue worth filing or fixing?

Reply:
Great catch. I've rewritten the multi-line deviation to start at the start column of the begin marker comment, and end at the end column of the end marker comment.